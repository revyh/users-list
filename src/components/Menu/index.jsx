import React from 'react';
import {Link} from 'react-router-dom';
import IconButton from 'material-ui/IconButton';
import MenuIcon from 'material-ui/svg-icons/navigation/menu';
import MenuItem from 'material-ui/MenuItem';
import Drawer from 'material-ui/Drawer';

const buttonStyles = {
  position: 'fixed',
  left: '5px',
  top: '5px',
};

export default class Menu extends React.Component {
  state = {
    isOpen: false,
  };

  handleMenu = isOpen => this.setState({isOpen});
  handleMenuClose = () => this.setState({isOpen: false});
  handleMenuToggle = () => this.setState({isOpen: !this.state.isOpen});

  render() {
    return (
      <div>
        <IconButton
          style={buttonStyles}
          tooltip="Open menu"
          onTouchTap={this.handleMenuToggle}
        >
          <MenuIcon />
        </IconButton>
        <Drawer
          docked={false}
          width={200}
          open={this.state.isOpen}
          onRequestChange={this.handleMenu}
        >
          {this.props.links.map(
            ({route, name: linkName}) => (
              <Link key={route} to={route} style={{textDecoration: 'none'}}>
                <MenuItem onTouchTap={this.handleMenuClose}>
                  {linkName}
                </MenuItem>
              </Link>
            ),
          )}
        </Drawer>
      </div>
    );
  }
}
