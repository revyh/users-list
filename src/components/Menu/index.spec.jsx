import React from 'react';
import {shallow} from 'enzyme';
import Menu from '.';

function getLinks() {
  return [
    {route: '/some-route', name: 'Some Name'},
    {route: '/another-route', name: 'Another Name'},
  ];
}

it('should render', () => {
  expect(shallow(<Menu links={getLinks()} />)).toMatchSnapshot();
});

it('should render in opened state', () => {
  const menu = shallow(<Menu links={getLinks()} />);

  menu.setState({isOpen: true});
  expect(menu).toMatchSnapshot();
});

it('should toggle menu when icon button is clicked', () => {
  const menu = shallow(<Menu links={getLinks()} />);

  menu.children('IconButton').simulate('touchTap');

  expect(menu.children('Drawer').prop('open')).toBeTruthy();
});

it('should close menu when menu item is clicked', () => {
  const menu = shallow(<Menu links={getLinks()} />);

  menu.children('IconButton').simulate('touchTap');
  menu.find('MenuItem').first().simulate('touchTap');

  expect(menu.children('Drawer').prop('open')).toBeFalsy();
});

it('should close menu when clicked outside opened menu', () => {
  const menu = shallow(<Menu links={getLinks()} />);

  menu.children('IconButton').simulate('touchTap');
  menu.children('Drawer').simulate('requestChange');

  expect(menu.children('Drawer').prop('open')).toBeFalsy();
});
