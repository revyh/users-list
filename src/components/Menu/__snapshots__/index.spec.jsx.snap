// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onTouchTap={[Function]}
    style={
      Object {
        "left": "5px",
        "position": "fixed",
        "top": "5px",
      }
    }
    tooltip="Open menu"
    tooltipPosition="bottom-center"
    touch={false}
  >
    <NavigationMenu />
  </IconButton>
  <Drawer
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={false}
    openSecondary={false}
    swipeAreaWidth={30}
    width={200}
    zDepth={2}
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/some-route"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        Some Name
      </MenuItem>
    </Link>
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/another-route"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        Another Name
      </MenuItem>
    </Link>
  </Drawer>
</div>
`;

exports[`should render in opened state 1`] = `
<div>
  <IconButton
    disableTouchRipple={false}
    disabled={false}
    iconStyle={Object {}}
    onTouchTap={[Function]}
    style={
      Object {
        "left": "5px",
        "position": "fixed",
        "top": "5px",
      }
    }
    tooltip="Open menu"
    tooltipPosition="bottom-center"
    touch={false}
  >
    <NavigationMenu />
  </IconButton>
  <Drawer
    disableSwipeToOpen={false}
    docked={false}
    onRequestChange={[Function]}
    open={true}
    openSecondary={false}
    swipeAreaWidth={30}
    width={200}
    zDepth={2}
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/some-route"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        Some Name
      </MenuItem>
    </Link>
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/another-route"
    >
      <MenuItem
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "top",
          }
        }
        checked={false}
        desktop={false}
        disabled={false}
        focusState="none"
        insetChildren={false}
        onTouchTap={[Function]}
        targetOrigin={
          Object {
            "horizontal": "left",
            "vertical": "top",
          }
        }
      >
        Another Name
      </MenuItem>
    </Link>
  </Drawer>
</div>
`;
